libraries/RC_Channel/RC_Channel.cpp

```diff
@@ -904,6 +904,26 @@ void RC_Channel::do_aux_function(const aux_func_t ch_option, const aux_switch_po
         break;
 #endif // HAL_MINIMIZE_FEATURES
 
+    case AUX_FUNC::GOPRO_MODE_TOGGLE: {
+        // Momentary toggle for changing GoPro capture modes
+        AP_Camera *camera = AP_Camera::get_singleton();
+        if (camera == nullptr) {
+            break;
+        }
+        switch (ch_flag) {
+        case LOW:
+            // nothing
+            break;
+        case MIDDLE:
+            // nothing
+            break;
+        case HIGH:
+            camera->gopro_capture_mode_toggle();
```

Comment:
This interface confuses me a little.  Why a "toggle" rather than just using low/middle/high?  I know my transmitter has many more modal switches than toggle switches!

Reply:
In the use case I'm developing this around, momentary buttons are aplenty and there are no real toggle or 3 way switches (a herelink controller). I did it as a 3-way originally and it was torture in actual operation trying to count 1.5 seconds holding a button down and letting go at just the right time to get video mode instead of camera mode.  Momentary triggering a cycle between modes was orders of magnitude more usable in reality.  In the case of a traditional RC with an array of latching switches instead of momentary buttons, I acknowledge this is less than perfect.